### DISCUSSION TITLE
How to do a analysis of model?

### DISCUSSION THREAD

**Discussion by jtyang-chem** [deepmodeling/deepmd-kit#1390] [Q&A] [answered]
How to do an analysis of models after training?
- I want to know which atomic energy contributes more to the total energy in some structures, but all parameters or middle results are wrapped in `*.pb` file, how to get them?
- Or, could I unwrap all the parameters? Is there any example or tutorial that I can follow?

**Comment by njzjz** [comment, marked as answer]
A python method is [provided here](https://docs.deepmodeling.org/projects/deepmd/en/v2.0.3/api_py/deepmd.infer.html#deepmd.infer.deep_pot.DeepPot.eval). You can get atomic energies by setting `atomic` to `True`.

To unwrap parameters, see https://github.com/deepmodeling/deepmd-kit/discussions/1271#discussioncomment-1635874.
